order-api/src/main/java/com/smoothstack/order/api/OrderApiController.java

```diff
@@ -37,8 +38,8 @@ public OrderApiController(NativeWebRequest request, OrderService orderService) {
 
     @Override
     @PreAuthorize("permitAll()")
-    public ResponseEntity<List<Order>> getOrder(@ApiParam(value = "if true only returns pending orders") @Valid @RequestParam(value = "active", required = false) Boolean active){
-        return orderService.getOrder(true);
+    public ResponseEntity<Order> getOrder(@ApiParam(value = "if true only returns pending orders") @Valid @RequestParam(value = "id", required = false) String id){
+        return orderService.getOrder(id);
```

Comment:
this should not be permitAll, at the very least it should be isAuthenticated(). PostAuthorize could be used to check if the order belongs to the authenticated user or if the user has admin authority
```
@PostAuthorize("hasAuthority('admin') or returnObject.getUser().getEmail() == authentication.principal.username ")
```
or do a custom authorization in the method. I think the order entity doesn't have a link to the user it belongs to right now, that will need added.

Reply:
I'll make a docker compose file for the microservices so that you can run the authentication server at the same time. This change can probably wait until then.